src/include/connection.h

```diff
@@ -255,6 +255,8 @@ struct __wt_name_flag {
         TAILQ_INSERT_HEAD(&(conn)->dhhash[bucket], dhandle, hashq);                              \
         ++(conn)->dh_bucket_count[bucket];                                                       \
         ++(conn)->dhandle_count;                                                                 \
+        WT_ASSERT(session, (dhandle)->type <= WT_DHANDLE_TYPE_TIERED_TREE);                      \
```

Comment:
I can see this assertion would fire if we were to add a new dhandle type after `WT_DHANDLE_TYPE_TIERED_TREE`. But one should not have to think about changing this code and if we didn't have the assertion, we wouldn't detect it. That's another argument for me to have a `WT_DHANDLE_TYPE_MAX`.

Reply:
Done. Replaced with WT_ASSERT(session, (dhandle)->type < WT_DHANDLE_TYPE_NUM);